### PR TITLE
add cd mxnet_lib/static and python/pypi stages to ci

### PR DESCRIPTION
**Pull request by mseth10**
## Description ##
Add CD "mxnet_lib/static" and "python/pypi" pipelines build and test stages to CI to enable catching issues before merging PRs.

The following CI stages have been added/removed.

- centos-cpu
  - (Added) CPU: CD Static Build
  - (Added) Python3: CentOS 7 CPU CD
  - (Added) PyPI package: CentOS 7 CPU CD
  - (Removed) Static build CPU CentOS7 Python with CMake
- centos-gpu
  - (Added) GPU: CD Static Build
  - (Added) Python3: CentOS 7 CPU CD
  - (Added) Quantization Python3: CentOS 7 GPU CD
  - (Added) PyPI package: CentOS 7 GPU CD
  - (Removed) Static build GPU CentOS7 Python with CMake

### CODE REVIEW

**Comment by mseth10** [comment]
@leezu please help review and merge. Thanks!

File: ci/jenkins/Jenkinsfile_centos_cpu [hunk 1]
```diff
@@ -37,14 +37,16 @@ core_logic: {
     custom_steps.compile_centos7_cpu('centos7_cpu'),
     custom_steps.compile_centos7_cpu_make('centos7_cpu_make'),
     custom_steps.compile_centos7_cpu_mkldnn(),
+    custom_steps.compile_static_cd_cpu('centos7_cpu_cd'),
```
Comment (leezu) [inline ci/jenkins/Jenkinsfile_centos_cpu:40]: The new test is a more elaborate version of `custom_steps.compile_static_python_cpu_cmake()` isn't it? If so, let's remove the  `custom_steps.compile_static_python_cpu_cmake()` in favor of the new approach.

Comment (mseth10) [reply ci/jenkins/Jenkinsfile_centos_cpu]: Done! Also removed custom_steps.compile_static_python_gpu_cmake() that was running cu92 in favor of new stage running cu102.

**Comment by mseth10** [comment]
@mxnet-bot run ci [windows-cpu]

**Comment by mseth10** [comment]
@leezu please help merge the pr. thanks!

**Comment by ChaiBapchya** [comment]
Curious why static builds are on Centos [and not on linux]
For instance for building binaries for AWS-MXNet we do it on linux 14.04 [and soon migrating to 16 or 18.04]

**Comment by leezu** [comment]
>  For instance for building binaries for AWS-MXNet we do it on linux 14.04 [and soon migrating to 16 or 18.04]

That would be a bad idea. You want to be compatible with https://www.python.org/dev/peps/pep-0599/ and for that you MUST build on CentOS7 or a system with equivalent glibc version

**Comment by ChaiBapchya** [comment]
Thanks for pointing it out.

**Pull request merged**